test/all.js

```diff
@@ -341,41 +341,42 @@ test('core caching after reopen regression', async function (t) {
   t.pass('did not infinite loop')
 })
 
-test('Corestore.from(corestore)', async function (t) {
-  const store = new Corestore(ram)
-  const ns1 = store.namespace('a')
-  const ns2 = store.namespace('b')
+test('session', async function (t) {
+  const store1 = new Corestore(ram)
+  const store2 = new Corestore(ram, { primaryKey: Buffer.alloc(32).fill('a') })
 
-  const store1 = Corestore.from(ns1, {
-    primaryKey: Buffer.alloc(32).fill('a')
-  })
-  const store2 = Corestore.from(ns2, {
-    primaryKey: Buffer.alloc(32).fill('a')
-  })
+  await store1.ready()
+  await store2.ready()
+
+  const session1 = store1.namespace('a').session()
```

Comment:
Hmm, this line highlights a potential issue. It's reasonable to expect that calling `session` on a namespaced corestore returns a session with the same namespace by default, which would mean passing `namespace: this._namespace` as a constructor opt.